pkg/capacityplanner/planner.go

```diff
@@ -0,0 +1,278 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package capacityplanner
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	genV1 "github.com/dell/csi-baremetal/api/generated/v1"
+	acrcrd "github.com/dell/csi-baremetal/api/v1/acreservationcrd"
+	accrd "github.com/dell/csi-baremetal/api/v1/availablecapacitycrd"
+	"github.com/dell/csi-baremetal/pkg/base/util"
+)
+
+// ACMap AC.Name to AC mapping
+type ACMap map[string]*accrd.AvailableCapacity
+
+// ACRMap ACR.Name to ACR mapping
+type ACRMap map[string]*acrcrd.AvailableCapacityReservation
+
+// SCToACMap SC name to ACMap mapping
+type SCToACMap map[string]ACMap
+
+// VolToACMap volume to AC mapping
+type VolToACMap map[*genV1.Volume]*accrd.AvailableCapacity
+
+// VolToACListMap volume to AC list mapping
+type VolToACListMap map[*genV1.Volume][]*accrd.AvailableCapacity
+
+// VolumesPlanMap NodeID to VolToACMap mapping
+type VolumesPlanMap map[string]VolToACMap
+
+// NodeCapacityMap NodeID to ACMap mapping
+type NodeCapacityMap map[string]ACMap
+
+// ACNameToACRNamesMap AC name to ACR names mapping
+type ACNameToACRNamesMap map[string][]string
+
+// CapacityReader methods to read available capacity
+type CapacityReader interface {
+	// ReadCapacity read capacity
+	ReadCapacity(ctx context.Context) ([]accrd.AvailableCapacity, error)
+}
+
+// ReservationReader methods to read capacity reservations
+type ReservationReader interface {
+	// ReadReservations read capacity reservations
+	ReadReservations(ctx context.Context) ([]acrcrd.AvailableCapacityReservation, error)
+}
+
+// CapacityPlaner describes interface for volumes placing planing
+type CapacityPlaner interface {
+	// PlanVolumesPlacing plan volumes placing on nodes
+	PlanVolumesPlacing(ctx context.Context, volumes []*genV1.Volume) (*VolumesPlacingPlan, error)
+}
+
+// CapacityManagerBuilder interface for capacity managers creation
+type CapacityManagerBuilder interface {
+	// GetCapacityManger returns CapacityManager
+	GetCapacityManger(logger *logrus.Entry, capReader CapacityReader) CapacityPlaner
+	// GetReservedCapacityManger returns ReservedCapacityManager
+	GetReservedCapacityManger(logger *logrus.Entry,
+		capReader CapacityReader, resReader ReservationReader) CapacityPlaner
+}
+
+// DefaultCapacityManagerBuilder is a builder for default CapacityManagers
+type DefaultCapacityManagerBuilder struct{}
+
+// GetCapacityManger returns default implementation of CapacityManger
+func (dcmb *DefaultCapacityManagerBuilder) GetCapacityManger(
```

Comment:
There is a typo: `Manger` -> `Manager`, same for ReservedCapacityManger.

Reply:
Fixed